tests/integration/models/applicationInvite.spec.ts

```diff
@@ -0,0 +1,130 @@
+// tslint:disable-next-line:import-blacklist
+import * as m from 'mochainon';
+import {
+	balena,
+	BalenaSdk,
+	givenAnApplication,
+	givenLoggedInUser,
+} from '../setup';
+const { expect } = m.chai;
+
+const TEST_EMAIL = 'user.test@example.org';
+const TEST_MESSAGE = 'Hey!, Join my app on balenaCloud';
+const TEST_ROLE = 'developer';
+
+describe('Application Invite Model', function() {
+	givenLoggedInUser(beforeEach);
+	givenAnApplication(beforeEach);
+
+	describe('given no application invite', function() {
+		describe('balena.models.applicationInvite.getAllByApplication()', function() {
+			it('shoud return an empty Array', function() {
+				const promise = balena.models.applicationInvite.getAllByApplication(1);
+				return expect(promise).to.become([]);
+			});
+
+			it('should support a callback with no options', function(done) {
+				// tslint:disable-next-line:ban-types
+				(balena.models.applicationInvite.getAllByApplication as Function)(
+					this.application.id,
+					function(
+						_err: Error,
+						applicationInvite: BalenaSdk.ApplicationInvite[],
+					) {
+						expect(applicationInvite).to.deep.equal([]);
+						return done();
+					},
+				);
+			});
+		});
+
+		describe('balena.models.applicationInvite.create()', function() {
+			it('should create and return an application invite', function() {
+				return balena.models.applicationInvite
+					.create(this.application.id, TEST_EMAIL, TEST_ROLE, TEST_MESSAGE)
+					.then(response => {
+						expect(response).to.have.property('id');
+						return balena.models.applicationInvite.getAllByApplication(
+							this.application.id,
+							{
+								$expand: {
+									is_invited_to__application: { $select: ['id'] },
+									invitee: { $select: ['email'] },
+									application_membership_role: { $select: ['name'] },
+								},
+								$select: ['message'],
+							},
+						);
+					})
+					.then(invites => {
+						expect(invites).to.have.length(1);
+						expect(invites[0].is_invited_to__application[0].id).to.equal(
+							this.application.id,
+						);
+						expect(invites[0].application_membership_role[0].name).to.equal(
+							TEST_ROLE,
+						);
+						expect(invites[0].message).to.equal(TEST_MESSAGE);
+					});
+			});
+		});
+	});
+
+	describe('given a single application invite', function() {
+		beforeEach(function() {
```

Comment:
Let's also make this a `before` and add a `[contained scenario]` note right above
> describe('given a single application invite [contained scenario]',